src/audio/google_rtc_audio_processing.c

```diff
@@ -0,0 +1,387 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Google LLC. All rights reserved.
+//
+
+#include <errno.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <google_rtc_audio_processing.h>
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/kpb.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/ipc/msg.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/notifier.h>
+#include <sof/lib/uuid.h>
+#include <sof/lib/wait.h>
+#include <sof/list.h>
+#include <sof/math/numbers.h>
+#include <sof/string.h>
+#include <sof/trace/trace.h>
+#include <sof/ut.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <user/trace.h>
+
+#define GOOGLE_RTC_AUDIO_PROCESSING_SAMPLERATE 48000
```

Comment:
question: how would this work if the input, e.g. a digital mic, is already downsampled to 16kHz, but the echo reference is indeed 48kHz?

Reply:
This is a use-case we do support however not all logic it already wired.
Same issue arise with number of channel: Playback is usually 2 but can be more sometimes or even mono. Mic could be one or more channel.
This is also something we do support however, this component is not ready yet for handling that.